ocsp/responder/multi_source.go

```diff
@@ -58,8 +60,39 @@ func (src *multiSource) Response(ctx context.Context, req *ocsp.Request) (*Respo
 		return nil, fmt.Errorf("looking up OCSP response for serial: %s err: %w", serialString, ctx.Err())
 
 	case primaryResult := <-primaryChan:
-		src.counter.WithLabelValues("primary_result").Inc()
-		return primaryResult.resp, primaryResult.err
+		// If there was an error requesting from the primary, don't bother
+		// waiting for the secondary, because we wouldn't be able to
+		// check the secondary's status against the (more reliable) primary's
+		// status.
+		if primaryResult.err != nil {
+			src.counter.WithLabelValues("primary_error").Inc()
+			return nil, primaryResult.err
+		}
+		// The primary response was fresh enough to serve, go ahead and serve it.
+		// TODO: use a clock.Clock
+		if time.Since(primaryResult.resp.ThisUpdate) < 60*time.Hour {
+			src.counter.WithLabelValues("primary_result").Inc()
+			return primaryResult.resp, nil
+		}
+
+		// The response was too stale to (ideally) serve. This will be a common
+		// path once we stop ocsp-updater from writing updated blobs to MariaDB.
+		// Try to serve from the secondary.
+		select {
+		case <-ctx.Done():
+			src.counter.WithLabelValues("timed_out_awaiting_secondary").Inc()
+			// Best-effort: return the primary response even though it's stale.
+			return primaryResult.resp, nil
+
+		case secondaryResult := <-secondaryChan:
+			if secondaryResult.err != nil {
+				src.counter.WithLabelValues("primary_error_secondary_error").Inc()
+				// Best-effort: return the primary response even though it's stale.
+				return primaryResult.resp, nil
+			}
+			src.counter.WithLabelValues("primary_error_secondary_success").Inc()
+			return secondaryResult.resp, nil
```

Comment:
Need to double-check that the secondary's status matches the primary's, since we have responses from both.